### ISSUE TITLE
Falling through sloped one-way platforms with Polygon/Edge colliders

### ISSUE THREAD

**Issue by 3noneTwo** [cjddmut/Unity-2D-Platformer-Controller#28]
While walking on one-way platforms, if the player walks up a sloped platform, they will immediately (within one or two frames) fall through it. This only happens if the platform uses a Polygon or Edge Collider. Box Colliders are fine, and walking down sloped platforms works correctly with all kinds of colliders.

 I've put together a small example scene using the stock PC2D Player prefab, [Tiled Map Editor](http://www.mapeditor.org/), and Sean Barton's [Tiled2Unity](http://www.seanba.com/Tiled2Unity).

**Project (Unity 5.1.1f1 / 2f1):** http://chicago-lollie.com/protoplatformer/git/SlopedOneWayPlatforms.rar

~~I'll note, I'm not sure if this is a PC2D issue or a Unity issue. Unity's [Manual entry for PlatformEffector2D](http://docs.unity3d.com/Manual/class-PlatformEffector2D.html) mentions a "Surface Arc" property, but this doesn't appear to be available in Unity 5.1.1f1. For all I know, that property could be the solution to this issue.~~

**Comment by 3noneTwo** [comment]
Unity 5.1.2f1 released literally the day after posting this issue, and "Surface Arc" now appears! However, its inclusion alone does not solve the issue.

**Comment by cjddmut** [comment]
Good catch, I'll take a look to see if a good resolution exists using the Unity components. I don't have time today to check but it may have something to do with the object 'up' and the normal of the surface not being perpendicular. 

I think the current behavior aligned itself with how Unity did work with one way platforms and they have resolved it with the 'Surface Arc' field you found. I'll have to add support for the Surface Arc in the motor. As far as the effectors go, the motor just handles them the same way as Unity but otherwise need to reimplement the behavior.

**Comment by astraldata** [comment]
Looks like it shouldn't be hard to solve -- take a peek at this post:

http://forum.unity3d.com/threads/platform-effector-2d-surface-arc-property.344624/

**Comment by cjddmut** [comment]
Definitely!

I'm swamped until the end of the month at which point I plan to finalize this version and start working on the next version (which will include this change).

**Comment by astraldata** [comment]
Any updates on this issue?

**Comment by cjddmut** [comment]
Not yet. I plan to address this when I start implementing the next version which I haven't had time to do yet. It's on the road map but I don't have a definite ETA on its fix.

**Comment by astraldata** [comment]
Just curious if you've made any headway on this. I'm not sure how big of a difference the new Unity method is to the current system you use internally (as I haven't looked at it in a while), otherwise I'd try to help with this.
